front/lib/api/assistant/actions/tables_query.ts

```diff
@@ -12,29 +12,43 @@ import type {
   TablesQueryOutputEvent,
   TablesQueryStartedEvent,
 } from "@dust-tt/types";
-import { BaseAction, Ok } from "@dust-tt/types";
+import {
+  BaseAction,
+  getTablesQueryResultsFileAttachment,
+  getTablesQueryResultsFileTitle,
+  Ok,
+} from "@dust-tt/types";
+import { stringify } from "csv-stringify";
 
 import { runActionStreamed } from "@app/lib/actions/server";
 import { DEFAULT_TABLES_QUERY_ACTION_NAME } from "@app/lib/api/assistant/actions/names";
 import type { BaseActionRunParams } from "@app/lib/api/assistant/actions/types";
 import { BaseActionConfigurationServerRunner } from "@app/lib/api/assistant/actions/types";
 import { renderConversationForModelMultiActions } from "@app/lib/api/assistant/generation";
+import { internalCreateToolOutputCsvFile } from "@app/lib/api/files/tool_output";
 import { getSupportedModelConfig } from "@app/lib/assistant";
 import type { Authenticator } from "@app/lib/auth";
 import { AgentTablesQueryAction } from "@app/lib/models/assistant/actions/tables_query";
 import { cloneBaseConfig, DustProdActionRegistry } from "@app/lib/registry";
+import { FileResource } from "@app/lib/resources/file_resource";
 import { sanitizeJSONOutput } from "@app/lib/utils";
 import logger from "@app/logger/logger";
 
 // Need a model with at least 32k to run tables query.
 const TABLES_QUERY_MIN_TOKEN = 28_000;
 const RENDERED_CONVERSATION_MIN_TOKEN = 4_000;
 
+// Max number of lines from the CSV ouptut file that will be saved in the DB
+// and inlined in the rendered conversation.
+const MAX_SNIPPET_LINES_QUERY_RESULT_FILE = 128;
```

Comment:
"file that will be saved in the DB"

You mean, saved in the DB because it's rendered ?

Reply:
Actually saved on the TablesQueryAction. Initially wanted to save the snippet as a File too, but batch-downloading files is pretty slow and we don't want to introduce that in the `renderConversationForModel`